src/functionals/gga_c_pbe.jl

```diff
@@ -1,26 +1,36 @@
-struct PbeCorrelation{Tlda,CA} <:
-       Functional{:gga,:c} where {Tlda,CA<:ComponentArray{<:Number}}
-    parameters::CA
+struct PbeCorrelation{NT,Tlda,Id} <:
+    Functional{:gga,:c} where {NT<:NamedTuple,Tlda,Id<:Union{Symbol,Val}}
+    parameters::NT
     lda::Tlda
-    identifier::Symbol
+    identifier::Id
 end
-function PbeCorrelation(parameters::ComponentArray, lda=DftFunctional(:lda_c_pw))
+function PbeCorrelation(parameters::NamedTuple, lda=DftFunctional(:lda_c_pw))
     PbeCorrelation(parameters, lda, :gga_c_pbe_custom)
 end
-function PbeCorrelation(parameters::ComponentArray, identifier::Symbol)
+function PbeCorrelation(parameters::NamedTuple, identifier::Symbol)
     PbeCorrelation(parameters, DftFunctional(:lda_c_pw), identifier)
 end
 
 identifier(pbe::PbeCorrelation) = pbe.identifier
 parameters(pbe::PbeCorrelation) = pbe.parameters
-function change_parameters(pbe::PbeCorrelation, parameters::ComponentArray;
+function to_isbits(pbe::PbeCorrelation)
+    PbeCorrelation(pbe.parameters, to_isbits(pbe.lda), Val{pbe.identifier}())
+end
+function change_parameters(pbe::PbeCorrelation, parameters::NamedTuple;
                            keep_identifier=false)
     if keep_identifier
         PbeCorrelation(parameters, pbe.lda, pbe.identifier)
     else
         PbeCorrelation(parameters, pbe.lda)
     end
 end
+# Change functional parameters based on an array of values. Assumes consistent ordering.
```

Comment:
What do you mean by assumes consistent ordering ? I would not assume that the keys are in the same order ... as this cannot be enforced in practice.

Reply:
Also can we not use a `ComponentArray` here ?

Reply:
In any case I would settle on one way of getting and setting the parameters and avoid the duplication of the interface. Otherwise this may derail. DftFunctionals is anyway already too complicated for what it's doing, so if we need to refactor parts of it to make GPU work and reshuffle functionality, that's ok.